src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -125,40 +160,83 @@ type SemVerInfo =
     
     member x.AsString
         with get() = x.ToString()
+        
+    member x.Equals(y) =
+        x.Major = y.Major && x.Minor = y.Minor && x.Patch = y.Patch && x.Build = y.Build && x.PreRelease = y.PreRelease
 
     override x.Equals(yobj) = 
         match yobj with
-        | :? SemVerInfo as y -> 
-            x.Major = y.Major && x.Minor = y.Minor && x.Patch = y.Patch && x.PreRelease = y.PreRelease && x.Build = y.Build && x.BuildMetaData = y.BuildMetaData 
+        | :? SemVerInfo as y -> x.Equals(y)
         | _ -> false
     
-    override x.GetHashCode() = hash (x.Minor, x.Minor, x.Patch, x.PreRelease, x.Build)
+    override x.GetHashCode() = hash (x.Minor, x.Minor, x.Patch, x.Build, x.PreRelease)
+    
+    member x.CompareTo(y) =
+        let comparison =  
```

Comment:
comparing each part only once and using the result is more transparent (and faster)